### DISCUSSION TITLE
Question please!

### DISCUSSION THREAD

**Discussion by msmanui** [Jikoo/Regionerator#126] [Q&A]
Hi there.

I have an overworld (Size: 1.3 TB), how long should take to remove the unused regions aprox? I would use it with WorldGuard.
As far as i read, technically all the region has to be modifed in order to know at same point if it will be deleted or not. But 99% of the regions wont be modifed.

**Comment by Jikoo** [comment]
That depends heavily on a lot of settings and is dependent on the number of region files rather than total world size. In general, assuming that you're asking "how long will a cycle take to complete" then Regionerator will by default take 2 seconds + processing time per region. If you're only using WG or other async capable plugins, the processing does not take much additional time. If you're using a non-async-capable plugin, processing takes much longer, several orders of magnitude.

**Reply by msmanui** [reply]
But i mean, i have regions of 1 year old that has never been modifed, how those are somehow being deleted if never its modifeid?

**Reply by Jikoo** [reply]
Regionerator reads the region list from the world files. It is aware of all regions, modified or not.

**Reply by msmanui** [reply]
> Regionerator reads the region list from the world files. It is aware of all regions, modified or not.

So basically it would take "2 seconds + processing time per region * amount of regions " right?

**Reply by Jikoo** [reply]
`(2s + processing time) * region count` to complete a cycle, yes. That's assuming you're using default settings, which are admittedly aimed towards lower-end machines.

There's also the pre-delete gather period. See https://github.com/Jikoo/Regionerator/wiki/FAQ#why-is-it-taking-so-long-to-start-deleting-area and https://github.com/Jikoo/Regionerator/wiki/FAQ#how-can-i-force-deletion-to-run-early
